server/command_hooks.go

```diff
@@ -405,6 +417,38 @@ func getDialogWithIntroductionText(introductionText string) model.Dialog {
 	return dialog
 }
 
+func (p *Plugin) executeCommandInteractive(args *model.CommandArgs) *model.CommandResponse {
+
+	post := &model.Post{
+		ChannelId: args.ChannelId,
```

Comment:
Please add this to fix the second issue
```suggestion
		ChannelId: args.ChannelId,
		RootId:    args.RootId,
```

Reply:
I already tried this, but then, clicking the button will trigger an error in `http_hooks.go`: `createPost: Invalid RootId parameter`. And if I remove the `RootId`, the grouping is not respected anymore.

Reply:
Could you please double check? My code change is about the post created via the slash command, not when the button is pressed.

Reply:
I know, but in `http_hooks.go` I use the `request.PostId` as the new Post's RootId and when I change your line, it fails...

Reply:
> I know, but in `http_hooks.go` I use the `request.PostId` as the new Post's RootId and when I change your line, it fails...

I see. But this is actually not correct. If the post with the buttons is not a root post, we should not use it's id as RootId. Instead we should get the post with `request.PostId` and use **it's** rootId.

Reply:
Thanks, it fixed the issue !